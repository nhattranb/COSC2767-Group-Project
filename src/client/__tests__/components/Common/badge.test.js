/*
 *   RMIT University Vietnam
 *   Course: COSC2767 Systems Deployment and Operations
 *   Semester: 2024C
 *   Assessment: Assignment 2
 *   Author: Le Dinh Ngoc Quynh
 *   ID: s3791159
 *   Created  date: 17 Dec 2024
 *   Last modified: 17 Dec 2024
 */

import React from 'react';
import { render } from '@testing-library/react';
import Badge from "../../../app/components/Common/Badge";

describe('Badge component', () => {
    it('renders correctly with default props', () => {
        const { asFragment } = render(<Badge>Default Badge</Badge>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly with primary variant', () => {
        const { asFragment } = render(<Badge variant="primary">Primary Badge</Badge>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly with danger variant', () => {
        const { asFragment } = render(<Badge variant="danger">Danger Badge</Badge>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly with custom className', () => {
        const { asFragment } = render(<Badge className="custom-class">Custom Class Badge</Badge>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly with borderless and round props', () => {
        const { asFragment } = render(<Badge borderless round={10}>Borderless Round Badge</Badge>);
        expect(asFragment()).toMatchSnapshot();
    });
});
