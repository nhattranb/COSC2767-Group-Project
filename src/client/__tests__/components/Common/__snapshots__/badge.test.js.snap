// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component renders correctly with borderless and round props 1`] = `
<DocumentFragment>
  <span
    class="custom-badge custom-badge-secondary"
    style="border-radius: 0;"
  >
    Borderless Round Badge
  </span>
</DocumentFragment>
`;

exports[`Badge component renders correctly with custom className 1`] = `
<DocumentFragment>
  <span
    class="custom-badge custom-class custom-badge-secondary"
    style="border-radius: 3px;"
  >
    Custom Class Badge
  </span>
</DocumentFragment>
`;

exports[`Badge component renders correctly with danger variant 1`] = `
<DocumentFragment>
  <span
    class="custom-badge custom-badge-danger"
    style="border-radius: 3px;"
  >
    Danger Badge
  </span>
</DocumentFragment>
`;

exports[`Badge component renders correctly with default props 1`] = `
<DocumentFragment>
  <span
    class="custom-badge custom-badge-secondary"
    style="border-radius: 3px;"
  >
    Default Badge
  </span>
</DocumentFragment>
`;

exports[`Badge component renders correctly with primary variant 1`] = `
<DocumentFragment>
  <span
    class="custom-badge custom-badge-primary"
    style="border-radius: 3px;"
  >
    Primary Badge
  </span>
</DocumentFragment>
`;
